scripts/tests/process-content-links.test.mjs

```diff
@@ -79,3 +74,55 @@ describe("processContentLinks - link mapping transformations", () => {
     expect(result).toContain(expected);
   });
 });
+
+describe("processContentLinks - relative link adjustment based on file type", () => {
+  const testCases = [
+    {
+      name: "should NOT adjust relative links for README.md files",
+      input: "See [guide](../docs/intro) and [local](./file).",
+      file: "README.md",
+      expectedContains: ["](../docs/intro)", "](./file)"],
+      expectedNotContains: ["](../../docs/intro)", "](../file)"],
+    },
+    {
+      name: "should adjust ../ to ../../ for non-README files",
+      input: "See [guide](../docs/intro).",
+      file: "user-guide/getting-started.md",
+      expectedContains: ["](../../docs/intro)"],
+      expectedNotContains: ["](../docs/intro)"],
+    },
+    {
+      name: "should adjust ./ to ../ for non-README files",
+      input: "See [local](./file).",
+      file: "tutorials/basic.md",
+      expectedContains: ["](../file)"],
+      expectedNotContains: ["](./file)"],
+    },
+    {
+      name: "should adjust both ./ and ../ for non-README files",
+      input: "See [guide](../docs/intro) and [local](./file).",
+      file: "actions/mutate.md",
+      expectedContains: ["](../../docs/intro)", "](../file)"],
+      expectedNotContains: ["](../docs/intro)", "](./file)"],
+    },
+    {
+      name: "should NOT adjust relative links in nested README.md",
+      input: "See [guide](../intro) and [local](./setup).",
+      file: "user-guide/README.md",
+      expectedContains: ["](../intro)", "](./setup)"],
+      expectedNotContains: ["](../../intro)", "](../setup)"],
+    },
+  ];
+
+  it.each(testCases)("$name", ({ input, file, expectedContains, expectedNotContains }) => {
+    const result = processContentLinks(input, file);
```

Comment:
This is a good example of a very clear test case where we provide inputs to code that we control and receive expected output. If I found myself editing the internals of `processContentLinks()` I'd know that this test would be a source of confidence that my code changes didn't break any expected behavior.